### ISSUE TITLE
Auto-save row outputs on row completion

### ISSUE THREAD

**Issue by gemmaguest** [mantidproject/mantid#34724]
In the ISIS Reflectometry interface, outputs for the groups and rows in the Runs table can be auto-saved if the relevant check boxes are ticked on the Save ASCII tab. The rows are only saved when the entire group finishes processing. Instead, users would like them to be saved as soon as the individual row finishes processing. 

I think this should be straightforward to do - the `BatchJobManager::getWorkspacesToSave` functions can be adjusted to always output the workspaces for rows if `includeRows` is true. Outputting the rows for a group can then be removed because the individual rows should already have been output (we don't want to re-save them if already done). We should test this carefully though in case I've misunderstood something - it relies on the notifications coming through in the right order for the right items in the table etc.

**Comment by rbauststfc** [comment]
There would be an added complication with this now that we support auto-saving multiple rows to a single ORSO file. It would be worth double-checking if this is still required before starting to look at it.
